opensearch/setup_models.py

```diff
@@ -0,0 +1,519 @@
+#!/usr/bin/python3
+
+import urllib.request
+import ssl
+import json
+from pathlib import Path
+import logging
+import re
+import time
+import os
+import sys
+
+OPENSEARCH_URL = "https://localhost:9200"
+ARYN_STATUSDIR = Path("/usr/share/opensearch/data/aryn_status")
+SSL_CTX = ssl.create_default_context()
+SSL_CTX.check_hostname = False
+SSL_CTX.verify_mode = ssl.CERT_NONE
+ML_CONFIG_FILE = ARYN_STATUSDIR / "ml_config.json"
+
+if int(os.environ.get("DEBUG", 0)) > 0:
+    logging.basicConfig(stream=sys.stdout, level=logging.DEBUG)
+else:
+    logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+
+
+def os_request_to_json(endpoint: str, body=None, method="GET"):
+    """
+    Send a request to opensearch.
+    endpoint is just the bit after the opensearch url
```

Comment:
"""We avoid using the opensearchpy library to avoid pulling in python dependencies."""